### PR TITLE
Add queue message model

### PR DESCRIPTION
**Pull request by mdayican**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/BPS-1085

### Change description ###

add the model for incoming queue message,
copy from processor.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/notificationservice/model/servicebus/in/NotificationMsg.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class NotificationMsg {
+
+    public final String zipFileName;
+    public final String jurisdiction;
+    public final String poBox;
+    public final String documentControlNumber;
+    public final ErrorCode errorCode;
```
Comment (mdayican) [inline src/main/java/uk/gov/hmcts/reform/notificationservice/model/servicebus/in/NotificationMsg.java]: do we really need error code?

File: src/main/java/uk/gov/hmcts/reform/notificationservice/model/servicebus/in/ErrorCode.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package uk.gov.hmcts.reform.notificationservice.model.servicebus.in;
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/notificationservice/model/servicebus/in/ErrorCode.java]: perhaps

```suggestion
package uk.gov.hmcts.reform.notificationservice.model.common;
```

? can be used for outgoing request too. what do you think?

File: build.gradle [hunk 3]
```diff
@@ -120,7 +120,7 @@ sonarqube {
     property "sonar.projectName", "Reform :: reform-scan-notification-service"
     property "sonar.projectKey", "uk.gov.hmcts.reform:reform-scan-notification-service"
     property "sonar.coverage.jacoco.xmlReportPaths", "${jacocoTestReport.reports.xml.destination.path}"
-    property "sonar.exclusions", "**/config/**"
+    property "sonar.exclusions", "**/config/**,**/model/**"
```
Comment (doncem) [inline build.gradle]: just a note: can be removed once tests are present

File: src/main/java/uk/gov/hmcts/reform/notificationservice/model/servicebus/in/NotificationMsg.java (hunk 1 above)
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/notificationservice/model/servicebus/in/NotificationMsg.java]: are values defined by provider?
making enum cause coupling to consumers.

File: src/main/java/uk/gov/hmcts/reform/notificationservice/model/servicebus/in/ErrorCode.java (hunk 2 above)
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/notificationservice/model/servicebus/in/ErrorCode.java]: may be we put enum and some other data in common but this message  is specific .
for message
uk.gov.hmcts.reform.notificationservice.model.request.incoming might be better?
uk.gov.hmcts.reform.notificationservice.model.common for Errorcode ?

File: build.gradle (hunk 3 above)
Comment (mdayican) [reply build.gradle]: yeep

File: src/main/java/uk/gov/hmcts/reform/notificationservice/model/servicebus/in/NotificationMsg.java (hunk 1 above)
Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/notificationservice/model/servicebus/in/NotificationMsg.java]: the specs were written by us and provider implemented this way

File: src/main/java/uk/gov/hmcts/reform/notificationservice/model/servicebus/in/ErrorCode.java (hunk 2 above)
Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/notificationservice/model/servicebus/in/ErrorCode.java]: or we can review later ;). can always refactor

**Pull request merged**